src/tools/gravity-parseList.c

```diff
@@ -45,6 +39,96 @@
 // Number of invalid domains to print before skipping the rest
 #define MAX_INVALID_DOMAINS 5
 
+// Validate domain name
+static inline bool __attribute__((pure)) valid_domain(const char *domain, const size_t len)
```

Comment:
There is no check for the overall and each label length (~255 chars and 63 chars I think.)

Reply:
But do we really need it? I think having an overall length check if meaningful but do we really need the individual labels one? In the end I don't think we need to be overly careful (remember, domains that don't exist also aren't harmful and don't cause extra time due to the logarithmic scaling of the tree-based lookup). However, ensuring that users do not pick up megabytes-long domains does make sense.

Reply:
The latest commit adds total and individual maximum and minimum length checks.

Reply:
I'd put in a limit. Been reading too much about buffer overflow attacks.

Reply:
Buffer overflows cannot happen in this code, when the line is too long to get enough memory for it, the process is aborted.